Improved Login experience, error messages, and published size.

Mbucari opened the pull request:
## [Update Dependencies](https://github.com/rmcrackan/Libation/pull/1406/commits/f98adef9e9194ddd45a0dab186fbae9a765967ac)
- Remove package references that are already included transitively
- Change Avalonia.ReactiveUI to ReactiveUI.Avalonia

## [Use xplat webview control for Audible login](https://github.com/rmcrackan/Libation/pull/1406/commits/fa238a09150d17f6da5b4850f44690d15859e79e)
- Use Avalonia-based webview control for Audible login with Chardonnay
- Remove webview interfaces from IInteropFunctions
- Remove Microsoft.Web.WebView2 package from WindowsConfigApp
- Add Microsoft.Web.WebView2 to LibationWinForms
- Remove all other login forms except the external login dialog (fallback in case webview doesn't work). The AudibleApi login with username/password doesn't work anymore. Need to use external browser login method.

## [Replace NPOI excel workbook library with ClosedXML](https://github.com/rmcrackan/Libation/pull/1406/commits/9533f80e89e5875c60e78fc3f9e042b8a1033d4d)
Reduce build bundle sizes by 30-40 MB. NPOI had a LOT of dependencies which were bloating Libation. [ClosedXML](https://github.com/ClosedXML/ClosedXML) is actively developed and has a very similar API surface.

## [Add multiple image sizes in windows folder icons](https://github.com/rmcrackan/Libation/pull/1406/commits/a34efb5e61143aeffc2f3459613c5914e3264e25)

The .ico format supports multiple sizes, and the image quality is usually much better if you resize the image yourself using a quality resampler than if Windows does it on the fly.

## [Improve error messages for Chardonnay](https://github.com/rmcrackan/Libation/pull/1406/commits/19efa8c918dc2fceb5a69a72d3ba3ff57fc07cc9)

- A message box alert should be possible regardless of the error
- If crash pre-logging, attempt to write to last used Libation log file

rmcrackan commented:
Holy cow this is a big change. I trust you but I admit to being nervous. This changes some login things which have been stable since practically day 1. It looks like it's gotta be simpler though. That's a ton of winform files deleted.

I certainly don't mind the size reduction that getting rid of N-POI gives us. I'm sure it was overkill for just creating a simple excel dump. Good find.

Good idea with utilizing the mult-sizes inside ico.s.

The next .net LTS is just around the corner. Anything you're looking forward to?

rmcrackan commented:
> We're going to be able to remove so much boilerplate.

A man after my own heart. I too get extra excited about things I can get rid of!

Mbucari commented:
> Or with vs without 2FA.

My main account has 2fa, so every test includes testing 2fa.

>  What about different regions?

I have an Italy account which I just tested, and the login page is slightly different so my javascript with inserts the username didn't work properly. I fixed it.

rmcrackan commented:
I'm about to head out. I have a uk account we can test with. I'll find the login tomorrow.

Mbucari commented:
I'm done. Ready whenever.

Mbucari commented:
## [Add Account column](https://github.com/rmcrackan/Libation/pull/1406/commits/5634fee2aa63c076cdb51f0e970b44974ae306e6) (#1398)

## [Improve Column Context Menu CheckBox display](https://github.com/rmcrackan/Libation/pull/1406/commits/01b5c18b2b25d27dd1bf2de95142273b59b1bedf)

The column visibility context menu checkboxes on Chardonnay were really tiny and hard to hit.

## [Add mock LibraryBook and Configuration capabilities](https://github.com/rmcrackan/Libation/pull/1406/commits/0038b2b5c1e924ad37ce010d3a31a68af784c806)

- Added `MockLibraryBook` which contains factories for easily creating mock LibraryBooks and Books
- Added mock Configuration
  - New `IPersistentDictionary` interface
  - New `MockPersistentDictionary` class which uses a `JObject` as its data store
  - Added `public static Configuration CreateMockInstance()`
    - This method returns a mock Configuration instance **and also sets the `Configuration.Instance` property**
    - Throws an exception if not in debug
- Updated all chardonnay controls to use the mocks in design mode. Previously I was using my actual database and settings file, but that approach is fragile and is unfriendly towards anyone else trying to work on it.

Mbucari commented:
I know I said I was done before, but now I'm really done.

rmcrackan commented:
Released v12.7!
